File: src/main/java/com/box/BoxTester.java
package com.box;

import java.util.Scanner;

public class BoxTester {
    public static void main(String[] args) {
        Box box3 = new Box3();
        Box box5 = new Box5();

        Scanner scanner = new Scanner(System.in);
        System.out.println("Please enter object's 「length」:");
        float length = scanner.nextFloat();
        System.out.println("Please enter object's 「width」:");
        float width = scanner.nextFloat();
        System.out.println("Please enter object's 「height」:");
        float height = scanner.nextFloat();

        if (box3.validate(length, width, height)) {
            System.out.println("Box3");
        } else if (box5.validate(length, width, height)) {
            System.out.println("Box5");
        } else {
            System.out.println("No suitable box ...");
        }
    }
}

File: src/main/kotlin/com/box/BoxTester.kt
package com.kotlin.box

import java.util.*


fun main() {
    val scanner = Scanner(System.`in`)
    println("Please enter object's 「length」:")
    val length = scanner.nextFloat()
    println("Please enter object's 「width」:")
    val width = scanner.nextFloat()
    println("Please enter object's 「height」:")
    val height = scanner.nextFloat()

    val box3 = Box3()
    val box5 = Box5()
    /*if (box3.validate(length, width, height)) {
        println("Box3")
    } else if (box5.validate(length, width, height)) {
        println("Box5")
    } else {
        println("No suitable box ...")
    }*/
    when {
        box3.validate(length, width, height) -> println("Box3")
        box5.validate(length, width, height) -> println("Box5")
        else -> println("No suitable box ...")
    }
}

abstract class Box {
    abstract fun validate(length: Float, width: Float, height: Float): Boolean
}

class Box3 : Box() {
    override fun validate(length: Float, width: Float, height: Float): Boolean {
        return length < 23f && width < 14f && height < 13f
    }
}

class Box5 : Box() {
    override fun validate(length: Float, width: Float, height: Float): Boolean {
        return length < 39.5f && width < 27.5f && height < 23f
    }

}



